pkg/dio/multi.go

```diff
@@ -192,6 +192,11 @@ func (t *multiReader) Read(p []byte) (int, error) {
 		}
 	}
 
+	// return 0, nil if there are no readers
+	if len(t.readers) == 0 {
```

Comment:
Hm.. not sure if I follow. The only way you get out of the for loop above is if `t.readers != 0 || t.err != nil`. The `t.err != nil` is handled right after the for loop. That leaves `t.readers != 0`, in which case you go through the other for loop at line 174, at the end of which isn't `t.readers != 0`? Or am I seeing things?